src/utils/getFilteredPeople.tsx

```diff
@@ -0,0 +1,64 @@
+import { Person } from '../types';
+
+const ZERO_YEARS = 1500 | 1600 | 1700 | 1800 | 1900 | 2000;
+
+export const getFilteredPeople = (
+  query: string,
+  sex: string,
+  centuries: string[],
+  people: Person[],
+): Person[] => {
+  let filteredPeople = [...people];
+
+  if (query) {
+    filteredPeople = filteredPeople.filter(person => {
+      if (person.name.toLowerCase().includes(query.toLowerCase())) {
+        return person;
+      }
+
+      if (
+        person.motherName &&
+        person.motherName.toLowerCase().includes(query.toLowerCase())
+      ) {
+        return person;
+      }
+
+      if (
+        person.fatherName &&
+        person.fatherName.toLowerCase().includes(query.toLowerCase())
+      ) {
+        return person;
+      }
+
+      return;
+    });
+  }
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter(person => {
+      if (person.sex === sex) {
+        return person;
+      }
+
+      return;
+    });
+  }
+
+  if (centuries.length) {
+    filteredPeople = filteredPeople.filter(person => {
+      const bornCenturyNumber =
+        person.born === ZERO_YEARS ? person.born : person.born + 100;
```

Comment:
The logic for determining the `bornCenturyNumber` is incorrect. The expression `person.born === ZERO_YEARS ? person.born : person.born + 100` does not correctly calculate the century. Consider revising this logic to correctly determine the century based on the year of birth.